vscode-trace-extension/src/utils/backend-tsp-client-provider.ts

```diff
@@ -41,7 +41,27 @@ export const addTspClientChangeListener = (listenerFunction: (tspClient: TspClie
 export async function isUp(): Promise<boolean> {
     const health = await getTspClient().checkHealth();
     const status = health.getModel()?.status;
-    return health.isOk() && status === 'UP';
+    const serverIsUp = health.isOk() && status === 'UP';
+    vscode.commands.executeCommand('setContext', 'traceViewer.serverOn', serverIsUp);
+    if (serverIsUp) {
+        // If the server is up, check for opened experiments
+        areExperimentsOpened();
```

Comment:
This, along with line 58 below in `areExperimentsOpened()`, which calls-back `isUp()`, seem to introduce a potential for an infinite loop, when the server is detected as being "up".

Reply:
addressed